modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2Cursor.java

```diff
@@ -46,12 +49,7 @@ public H2Cursor(GridCursor<H2Row> cursor) {
 
     /** {@inheritDoc} */
     @Override public Row get() {
-        try {
-            return cursor.get();
-        }
-        catch (IgniteCheckedException e) {
-            throw DbException.convert(e);
-        }
+        return cur;
```

Comment:
We could catch a situation where `.get()` is called before first `.next()`. previously, I guess `cursor.get()` would deal with this, now we silently return `null` while method is not `@Nullable`.

Reply:
It looks like there is no  special contract for this case for `Cursor` interface. If underlying cursor in the previous version returned 'null', the 'H2Cursor' would also return 'null' in this case. Also other H2's 'Cursor' implementations simply return 'null' values  in such cases.